core/src/main/scala/com/salesforce/op/stages/impl/preparators/SanityCheckerMetadata.scala

```diff
@@ -92,9 +92,8 @@ case class SanityCheckerSummary
     sample: Double
   ) {
     this(
-      correlationsWLabel = new Correlations(
-        stats.filter(s => s.corrLabel.isDefined && !s.corrLabel.get.isNaN).map(s => s.name -> s.corrLabel.get),
-        stats.filter(s => s.corrLabel.isDefined && s.corrLabel.get.isNaN).map(_.name),
+      correlations = new Correlations(
+        stats.filter(s => s.corrLabel.isDefined).map(s => (s.name, s.corrLabel.get, s.featureCorrs)),
```

Comment:
Do we not have to check for NaNs anymore? What if the feature is constant and we turn off the min variance threshold?

Reply:
So I fixed it - in that when it goes to metadata it turns it into strings and then when it gets out back into the class it is turned into doubles. Now nans work! The json serializer would not have helped because it was a problem with the spark metadata.